retrofit/src/main/java/retrofit2/Response.java

```diff
@@ -147,12 +147,23 @@ public boolean isSuccessful() {
     return rawResponse.isSuccessful();
   }
 
-  /** The deserialized response body of a {@linkplain #isSuccessful() successful} response. */
+  /**
+   * The deserialized response body of a {@linkplain #isSuccessful() successful} response.
+   *
+   * <p>If the body is of type {@link ResponseBody}, calling {@link ResponseBody#close close} is
```

Comment:
I don't think this is helpful because it splits the type system. 

By following this doc you need to differentiate between responses that are created from streaming endpoints and responses that are not. If you write general purpose code to accept a `Response`, you need to clarify whether it's a streaming one. Also if you add `@Streaming` you create a resource leak.

It's simpler and more resilient to document that this must be closed under all circumstances.

Reply:
Thanks for the reply. Yeah, that makes sense. So you are suggesting "If the body is of type ResponseBody, close() should always be called to avoid leaking resources"? Am I correct?

Reply:
@swankjesse  I've updated the PR, can you let me know if that looks correct, please?